src/v/cluster/cloud_metadata/offsets_recovery_manager.cc

```diff
@@ -56,7 +56,7 @@ ss::future<error_outcome> offsets_recovery_manager::recover(
           model::kafka_consumer_offsets_nt, parent_retry.get_deadline());
     }
     for (size_t i = 0; i < snapshot_paths_per_pid.size(); i++) {
-        auto pid = model::partition_id{i};
+        auto pid = model::partition_id(i);
```

Comment:
question for my understanding: whats the fix here? Doesn't list initialization prevent a narrowing conversion?

Reply:
> question for my understanding: whats the fix here? Doesn't list initialization prevent a narrowing conversion?

Yes, and clang-18 warns, and -Werror.

`partition_id::type` is `int32_t` so either it needs to be narrowed, or the returned `size_t` needs casting.